modules/basic-theme/src/Volo.Abp.AspNetCore.Components.Web.BasicTheme/Themes/Basic/FirstLevelNavMenuItem.razor

```diff
@@ -1,13 +1,19 @@
-﻿@using Volo.Abp.UI.Navigation
+﻿@using Volo.Abp.AspNetCore.Components.Web.BasicTheme.Navigation;
+@using Volo.Abp.UI.Navigation
 @{
     var elementId = MenuItem.ElementId ?? "MenuItem_" + MenuItem.Name.Replace(".", "_");
     var cssClass = string.IsNullOrEmpty(MenuItem.CssClass) ? string.Empty : MenuItem.CssClass;
     var disabled = MenuItem.IsDisabled ? "disabled" : string.Empty;
     var url = MenuItem.Url == null ? "#" : MenuItem.Url.TrimStart('/', '~');
+    var customComponentType = MenuItem.GetComponentTypeOrDefault();
 }
 @if (MenuItem.IsLeaf)
 {
-    if (MenuItem.Url != null)
+    if (customComponentType != null)
+    {
+        <DynamicComponent Type="@customComponentType" />
+    }
+    else if (url != null)
```

Comment:
is this mistake? the `url` will never `null`, thus the `else` block below will never reach.
revert to use `MenuItem.Url != null` ?

Reply:
Done in https://github.com/abpframework/abp/pull/15625